Libraries/xcassets/Headers/xcassets/Asset/Asset.h

```diff
@@ -29,6 +29,7 @@ class Asset {
 private:
     FullyQualifiedName         _name;
     std::string                _path;
+    libutil::Filesystem const *_filesystem;
```

Comment:
In that case, you can, but I can imagine other uses of the same objects that don't. For example, imagine if you made an asset catalog editor UI, you'd probably want to hold onto `xcassets::Asset::Asset` objects for a long time but the filesystem might be transient from the open dialog.

Of course it'd be possible to restructure that to work with this, but it requires understanding that implementation detail of the `xcassets::Asset::Asset` in order to use it safely. If the filesystem is passed in separately to the functions that need to use it, then you can't get it wrong.